backend/git_real/models.py

```diff
@@ -179,19 +178,35 @@ class Meta:
     @classmethod
     def create_or_update_from_github_api_data(cls, repo, request_body):
 
-        head_commit = request_body["head_commit"]
-        head_commit_url = head_commit["url"]
+        head_commit = request_body["head_commit"] or request_body['repository']
+        head_commit_url = head_commit["url"] or request_body['compare']
         ref = request_body["ref"]
-        defaults = {
-            "commit_timestamp": head_commit["timestamp"],
-            "author_github_name": head_commit["author"]["username"],
-            "committer_github_name": head_commit["committer"]["username"],
-            "message": head_commit["message"],
-            "pusher_username": request_body["pusher"]["name"],
-            "pushed_at_time": github_timestamp_int_to_tz_aware_datetime(
-                int(request_body["repository"]["pushed_at"])
-            ),
-        }
+
+        try:
+
+            defaults = {
+                "commit_timestamp": head_commit.get("timestamp"),
+                "author_github_name": head_commit.get("author").get("username"),
+                "committer_github_name": head_commit.get("committer").get("username"),
+                "message": head_commit.get("message"),
+                "pusher_username": request_body.get("pusher").get("name"),
+                "pushed_at_time": github_timestamp_int_to_tz_aware_datetime(
+                    int(request_body["repository"]["pushed_at"])
+                ),
+            }
+
+        except (TypeError, AttributeError):
+
+            defaults = {
+                "commit_timestamp": head_commit.get('updated_at'),
+                "author_github_name": request_body.get("pusher").get("email"),
```

Comment:
```suggestion
            defaults = {
                "commit_timestamp": head_commit.get("updated_at"),
                "author_github_name": request_body.get("pusher").get("email"),
```

Reply:
Hello

I made the requested changes but there is one thing I am not yet sure of.  Once this 'bug' is fixed I don't think we are going to have the two added assets I put in there and we also won't need the test I added either.  For now they are simply there to show that the fix works for the payloads we've been getting from Github's webhook where the payloads did not contain the data we needed.

So in all, Sheena should confirm, but I think that once we are happy that we have a work around for this we can get rid of the added assets and the test?